### ISSUE TITLE
Merge Student into newly created Offering when MASS offering changes are approved

### ISSUE THREAD

**Issue by JasonCTang** [bcgov/SIMS#2040]
**User story**
As a ministry user, I want an offering that has been changed with the mass offering change function to be set to approved status even if it no longer aligns with its parent program. All **completed** applications attached to that offering should be be reassessed. This is important as it changes the negative impact to the ministry user instead of the student when the ministry user fails to properly review the mass change for that offering to ensure compliance.

**Acceptance Criteria**

- [ ] Ensure the offering is placed into the approved status if approved by the ministry user during change process. This should occur even if the offering will no longer align with its parent program.
- [ ] Validations that leads to an offering being 'Pending' for Ministry approval will be bypassed.

**Business context**
Original outline had these scenarios leading to the application being cancelled due to the miss match. We no longer want this to occur and would prefer that the student not be punished for ministry review errors. The ministry user should be catching this miss match before the application is approved. Therefore when they approve the mass offering change in these situations the system will place it into approved status. 

Image below shows examples of parts of the offering creating a miss match. Intended path is for these to be reviewed and rejected by min user if not desired. 

Please note additional ticket #3123 addresses future update in 2.x where mass offering changes will also reassess applications that have reached assessed or later in the process.

<img width="879" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNmVBQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--857de359f03e7d7632b0a5479ec2cc63b7a95a81/image.png" alt="image.png" />

**Comment by CarlyCotton** [comment]
@HRAGANBC @Joshua-Lakusta 
I think we need to make sure our content in SIMS, the instructional guide and ops manual is **very clear** about the consequences of this action before it goes off to 2x/3x. The instructional guide doesn't mention anything beyond editing is limited so as not to impact active applications.

If we want to change the behaviour to something else then I'd support this being an enhancement at 2x/3x.

**Comment by michesmith** [comment]
@Joshua-Lakusta Is this a bug- its not a bug- regardless this ticket needs to be adjusted to reflect if its a bug or an enhancement- Thank you (smiley face)

**Comment by Joshua-Lakusta** [comment]
@HRAGANBC @CarlyCotton I am confused at the desired state for this. Ideally we would want to reassess any submitted applications correct? We treat submitted applications as live applications while doing program year limits etc. So from a consistency perspective should the stage at which our system starts doing all the reassessments be the line for when this should reassess? Either way this is not a bug. However, the desired state should be discussed. Rewrote the ticket as a user story. Also feeling that this is linked to ticket 2894 (retested) where the mass offering change is not functioning.

**Comment by Joshua-Lakusta** [comment]
@HRAGANBC Ticket updated to a user story around our discussion earlier. Please review when able.

**Comment by Joshua-Lakusta** [comment]
@HRAGANBC New version rebuilt based on our newest recent conversation. Take a quick peak because i would love to take it to Lather.

**Comment by HRAGANBC** [comment]
@andrewsignori-aot to test this out and get back with current behaviour and whether ticket needs to be pursued as stated

**Comment by andrewsignori-aot** [comment]
@HRAGANBC @Joshua-Lakusta during "DEV Dress" we were able to see that the functionality is already worked as described in the first AC.
A bulk offering request with a waring message will still be sent for Ministry bulk offering approval, and, once approved, will have its staus changed to `Approved`.

**Comment by Joshua-Lakusta** [comment]
After discussion with Devs. Now that mass change of offering functions I was able to run this through more cleanly. After new tests with the understanding that only completed apps are reassessed. I was able to create a new program and offering both FT and then have 2 students attached to the offering. I then changed the offering to PT which created the miss match highlighted in the ticket. Warning banner appeared as desired. On approval in the ministry portal the approval can now get through without an unexpected error toast. The completed applications reassessed as expected and the offering moved straight in to approved status even though it does not match its parent program. All AC's of this ticket satisfied in this test. Discussion completed with assigned dev and I am closing the ticket. (Test captured in test quality as well for record keeping)
